### PR TITLE
Bump master to SE 11, introduce JPMS descriptors

### PR DESCRIPTION
**Pull request by lukasj**
This adds JPMS descriptors to the project and sets Java SE 11 as the minimal required Java SE version to build & run master. Note that running EclipseLink on the module path should still be considered as highly experimental.

### CODE REVIEW

**Review by rfelcman — COMMENTED** [review]
What about JPMS usage in the tests https://maven.apache.org/surefire/maven-surefire-plugin/examples/jpms.html ?
But it leads into additional changes in the `module-info.java` files.
E.g.
See my test in JPA.JPQL Maven module:
modified `jpa/org.eclipse.persistence.jpa.jpql/src/main/java/module-info.java`

```
module org.eclipse.persistence.jpa.jpql {

    requires transitive java.sql;

    exports org.eclipse.persistence.jpa.jpql;
    exports org.eclipse.persistence.jpa.jpql.parser;
    exports org.eclipse.persistence.jpa.jpql.tools;
    exports org.eclipse.persistence.jpa.jpql.tools.model;
    exports org.eclipse.persistence.jpa.jpql.tools.model.query;
    exports org.eclipse.persistence.jpa.jpql.tools.resolver;
    exports org.eclipse.persistence.jpa.jpql.tools.spi;
    exports org.eclipse.persistence.jpa.jpql.tools.utility;
    exports org.eclipse.persistence.jpa.jpql.tools.utility.iterable;
    exports org.eclipse.persistence.jpa.jpql.utility;
    exports org.eclipse.persistence.jpa.jpql.utility.iterable;
}
```

new one in `jpa/org.eclipse.persistence.jpa.jpql/src/test/java/module-info.java`
```
module org.eclipse.persistence.jpa.jpql.test {

    requires org.eclipse.persistence.jpa.jpql;
    requires jakarta.persistence;
    requires junit;
    exports org.eclipse.persistence.jpa.tests.jpql to junit;
}
```

**Comment by lukasj** [comment]
> What about JPMS usage in the tests?

that's for another round. There are issues which needs to be resolved first, ie moxy depends on "old" jmockit, some dependencies are not proper modules (yet) + there are also bugs in el codebase which have to be fixed (ie moxy should be ignoring `$VALUES` filed in enums) but I don't think it makes sense to do everything at once

File: bundles/eclipselink/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    requires static jakarta.annotation;
+    requires static jakarta.json;
+    requires static jakarta.mail;
+    requires static jakarta.persistence;
+    requires static jakarta.validation;
+    requires static jakarta.ws.rs;
+    requires static jakarta.xml.bind;
+
+    requires static jakarta.cdi; //AM
+    requires static jakarta.el; //AM
```
[the remark this answers is not in the document]
Comment (lukasj) [reply bundles/eclipselink/src/main/java/module-info.java:32]: What is the difference between travis, who can build it, and your environment? `Automatic-Module-Name` of `jakarta.el:jakarta.el-api:4.0.0` is set in its manifest to `jakarta.el`. Note that this is `static` dependency, so it is not required to be available on module-path at runtime

File: bundles/eclipselink/pom.xml [hunk 2]
```diff
@@ -357,49 +291,6 @@
             <scope>provided</scope>
         </dependency>
 
-        <dependency>
```
[the remark this answers is not in the document]
Comment (lukasj) [reply bundles/eclipselink/pom.xml:360]: no. There are multiple ways one can include them. This part of the change does the trick without explicit dependency on source artifacts:
```
-                           <includeClassifiers>sources</includeClassifiers>
+                           <classifier>sources</classifier>
```

File: bundles/eclipselink/pom.xml [hunk 3]
```diff
@@ -686,38 +584,4 @@
             </plugin>
         </plugins>
     </build>
-
-    <profiles>
-        <!-- sdo: api dependencies -->
```
[the remark this answers is not in the document]
Comment (lukasj) [reply bundles/eclipselink/pom.xml:691]: it's packaged in the `eclipselink.jar` and `org.eclipse.persistetce.sdo.jar` - without that, there is no way to make sdo stuff working on the JPMS properly without figuring out proper set of add-exports/add-reads/patch-module options and that would make it really painful to use it in other projects, like Metro

File: bundles/eclipselink/src/main/java/module-info.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rfelcman) [reply bundles/eclipselink/src/main/java/module-info.java:32]: Sorry my mistake, I had some strange build of `jakarta.el:jakarta.el-api:4.0.0` in my local Maven repo.

File: bundles/others/pom.xml [hunk 4]
```diff
@@ -358,6 +358,23 @@
             <artifactId>jakarta.xml.bind-api</artifactId>
             <classifier>sources</classifier>
         </dependency>
+
```
[the remark this answers is not in the document]
Comment (lukasj) [reply bundles/others/pom.xml:361]: only those which which were there except of nosql one

File: bundles/others/pom.xml [hunk 5]
```diff
@@ -442,31 +459,15 @@
...
-                    </execution>
+                    <!--Disabled by default-->
                     <execution>
                         <id>nosql-javadoc.jar</id>
                         <phase>prepare-package</phase>
                         <goals>
                             <goal>jar</goal>
                         </goals>
                         <configuration>
+                            <skip>true</skip>
```
[the remark this answers is not in the document]
Comment (lukasj) [reply bundles/others/pom.xml:470]: no, problem with that is that I haven't found a way to produce javadoc for 2+ modules through maven yet. I'm also not sure we need it.

**Review by rfelcman — APPROVED** [review]
LGTM

**Pull request merged**